cyclops-ctrl/internal/prometheus/handler.go

```diff
@@ -0,0 +1,37 @@
+package prometheusHandler
+
+import (
+	"os"
+
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var setupLog = ctrl.Log.WithName("setup")
+
+var moduleQueued = prometheus.NewGauge(prometheus.GaugeOpts{
+	Name:      "modules_deployed",
+	Help:      "No of modules Inc or Dec",
+	Namespace: "cyclops",
+})
+
+func init() {
+	if err := prometheus.Register(moduleQueued); err != nil {
+		setupLog.Error(err, "unable to connect prometheus")
+		os.Exit(1)
+	}
+}
```

Comment:
Can we make this a constructor and call it from main.go? You can then pass the setupLog not to duplicate it.

Reply:
Also, can you remove os.Exit(1) since not having is not a critical error. We can make this configurable later on

Reply:
Can you please explain me what exactly I have to do, I'm a bit confused!

Reply:
I would like to turn `init` function into a constructor that returns a `Monitor` struct that will have the Prometheus gauge as its property. Then init the gauge and register it to Prom.

```suggestion
type Monitor struct {
       ModulesDeployed prometheus.Gauge
}

func NewMonitor() (Monitor, error) {
        m := Monitor{
             ModulesDeployed: prometheus.NewGauge(prometheus.GaugeOpts{
	              Name:      "modules_deployed",
	              Help:      "No of modules Inc or Dec",
	              Namespace: "cyclops",
              })
        }

	if err := prometheus.Register(m.ModulesDeployed); err != nil {
		setupLog.Error(err, "unable to connect prometheus")
		return Monitor{}, err
	}
	
	return m, nil
}
```
Dont judge, I coded this in a GitHub comment

Reply:
Oh but how would you call this in the `main` function

Reply:
you can do
```go
monitor, err := prometheus.NewMonitor(setupLog)
if err != nil {
  setupLog.Error(err, "failed to create monitor")
}
```

Reply:
Also how to handle the Inc and the Dec func?

Reply:
you can turn those into methods of Monitor
```go
func (m Monitor) IncModule() {
  m.ModulesDeployed.Inc()
}
```